csp_billing_adapter_amazon/plugin.py

```diff
@@ -84,5 +90,55 @@ def get_csp_name(config: Config):
 
 @csp_billing_adapter.hookimpl(trylast=True)
 def get_account_info(config: Config):
-    """Return a dictionary with account information"""
-    pass
+    """
+    Return a dictionary with account information
+
+    The information contains the metadata for document, signature and pkcs7.
+    """
+    account_info = _get_metadata()
+    account_info['document'] = json.loads(account_info.get('document', '{}'))
+    account_info['cloud_provider'] = 'amazon'
+
+    return account_info
+
+
+def _get_api_header():
+    """Get the header to be used in requests to the metadata service,
+        IMDs. Prefer IMDSv2 which requires a token."""
+    request = urllib.request.Request(
+        METADATA_TOKEN_URL,
+        headers={'X-aws-ec2-metadata-token-ttl-seconds': '21600'},
+        method='PUT'
+    )
+    try:
+        token = urllib.request.urlopen(request).read().decode()
+    except urllib.error.URLError:
+        return {}
+
+    return {'X-aws-ec2-metadata-token': token}
+
+
+def _get_metadata():
+    metadata_options = ['document', 'signature', 'pkcs7']
+    metadata = {}
+    request_header = _get_api_header()
+    for metadata_option in metadata_options:
+        metadata[metadata_option] = _fetch_metadata(
+            metadata_option,
+            request_header
+        )
+    return metadata
+
+
+def _fetch_metadata(uri, request_header):
+    """Return the response of the metadata request."""
+    url = METADATA_ADDR + '/dynamic/instance-identity/' + uri
+    data_request = None
+    value = b''
```

Comment:
superfluous assignment of `value`. If the url used for the creation of the `Request` object is not valid the `urlopen` will throw. Then `value` is not used. In any other case `value` will be assigned in the `try:` block and as such can be decoded.